modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/snapshot/storage/FeesAddressesStorage.scala

```diff
@@ -0,0 +1,10 @@
+package org.tessellation.node.shared.domain.snapshot.storage
+
+import org.tessellation.schema.GlobalSnapshotInfo
+import org.tessellation.schema.address.Address
+
+trait FeesAddressesStorage[F[_]] {
+  def setAllFeesAddresses(globalSnapshotInfo: GlobalSnapshotInfo): F[Unit]
+  def addMetagraphFeeAddresses(metagraphId: Address, addresses: List[Address]): F[Unit]
+  def getAllFeesAddresses: F[Map[Address, List[Address]]]
+}
```

Comment:
I think it can be reduced to:
```scala
trait FeeStorage[F[_]] {
  def set(globalSnapshotInfo: GlobalSnapshotInfo): F[Unit]
  def addMetagraph(metagraphId: Address, addresses: List[Address]): F[Unit]
  def getAll: F[Map[Address, List[Address]]]
}
```

As I see, we use `Address` for metagraph identifier. It may be worth creating an alias or newtype like:
```scala
@newtype
case class MetagraphId(address: Address)
```

Reply:
@marcinwadon about the alias - would be nice to see:
```scala
Map[MetagraphId, List[Address]]
```

But I am wondering how much refactoring it needs to change it everywhere :D